pkg/cmd/cli/secrets/unlink_secret_from_object.go

```diff
@@ -88,6 +88,7 @@ func (o UnlinkSecretOptions) unlinkSecretsFromServiceAccount(serviceaccount *kap
 			if secretname == serviceaccount.Secrets[i].Name {
```

Comment:
Also, we should switch the outer loop to iterate in order so we don't reverse the list when removing